### ISSUE TITLE
twisted.conch.test.test_recvline._StdioMixin.setUp runs child process with unportable parameters

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2977]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2977|
|Type|defect|
|Created|2008-01-05 16:46:47Z|
|Branch|https://github.com/twisted/twisted/tree/stdio-mixin-spawnprocess-2977|

The ```spawnProcess``` call in ```setUp``` has a couple problems:

  * ```argv[0]``` is hard-coded as _python2.3_.  It should just be ```sys.executable```.  This just mirrors the actual executable already passed to ```spawnProcess```.
  * The file to run, ```stdio.__file__```, passed as ```argv[1]``` may be a bytecode file, which Python won't run.

<details><summary>Searchable metadata</summary>

```
trac-id__2977 2977
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_stdio_mixin_spawnprocess_2977 branches/stdio-mixin-spawnprocess-2977
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1199551607000000 1199551607000000
changetime__1199632074000000 1199632074000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22273]) Branching to 'stdio-mixin-spawnprocess-2977'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Fixed.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

The copyright of the file should be updated. If you could add a module docstring, that would be great. You can merge whenever you want.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [22283]) Merge stdio-mixin-spawnprocess-2977

Author: exarkun
Reviewer: therve
Fixes [#2977](https://github.com/twisted/twisted/issues/2977)

Change the process spawning code in the insults test helper _StdioMixin so
that it provides the correct full path to the Python executable in argv[0]
and so that it does not try to run a bytecode file.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
